components/src/core/components/List/types.ts

```diff
@@ -0,0 +1,126 @@
+interface Visible {
+  visible?: boolean;
+}
+
+interface ListEmitEvent {
+  type: string;
+  identifier: string;
+}
+
+interface Tooltip {
+  tooltip?: string;
+  flow?: string;
+}
+
+interface Style {
+  [key: string]: string | undefined;
+}
+
+interface Icon {
+  iconName?: string;
+  "icon-size"?: string;
+  iconSize?: string;
+  iconStyle?: string;
+}
+
+interface ButtonIcon extends Icon {
+  "icon-name"?: string;
```

Comment:
Better to use a consistent naming convension.

Reply:
used camel case and made relevant reference changes on Vue App